libsplinter/src/biome/oauth/store/diesel/operations/update_session.rs

```diff
@@ -57,22 +61,71 @@ where
                             ),
                         ))
                     } else {
-                        // All checks have completed, update the entry
-                        update(
-                            oauth_user_sessions::table.filter(
-                                oauth_user_sessions::splinter_access_token
-                                    .eq(session.splinter_access_token()),
+                        // All checks have completed; remove the old entry and add the updated one.
+                        // We can't just update the existing entry because Diesel doesn't provide a
+                        // way to set the "last_authenticated" column to the default value
+                        // (database-assigned timestamp).
```

Comment:
Should use a trigger instead.

Reply:
Replaced the code change with migrations for adding a trigger.